modules/core/src/main/scala/compiler.scala

```diff
@@ -59,26 +59,23 @@ object QueryParser {
     }
   }
 
-  def parseOperation(op: Operation, fragments: Map[String, FragmentDefinition]): Result[UntypedOperation] = op match {
-    case Operation(opType, _, vds, _, sels) =>
-      val q = parseSelections(sels, None, fragments)
-      val vs = vds.map {
-        case VariableDefinition(nme, tpe, _) => UntypedVarDef(nme.value, tpe, None)
+  def parseOperation(op: Operation, fragments: Map[String, FragmentDefinition]): Result[UntypedOperation] = {
```

Comment:
Scala 3 warns that this single-case match would only fail with `case null =>` so I collapsed it to a destructuring assignment. There are a few more of these.